### PR TITLE
Set SameSite and Secure property for authentication cookie

### LINKED ISSUE #6039: [Enhancement]: Set SameSite and Secure property for authentication cookie
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description of problem

1. Defining SAMESITE to ensure that cookies are not sent to a third-party site or address. The value of this tag must be defined as "Strict".
2. Definition of the SECURE tag to ensure that the cookie is transmitted through a secure communication path and that it cannot eavesdrop (the SECURE tag should be activated after activating the HTTPS protocol).

### Description of solution

For the secure enabled, I think we should check SSL enabled in DNN and update the cookie when logging user.
for set SamaSite to Stric, we should just update the authentication cookie

### Description of alternatives considered

_No response_

### Anything else?

_No response_

### Do you be plan to contribute code for this enhancement?

- [X] Yes

### Would you be interested in sponsoring this enhancement?

- [ ] Yes

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by Mostafa-Moafi**
Fixes #6039

### CODE REVIEW

File: DNN Platform/Library/Security/PortalSecurity.cs [hunk 1]
```diff
@@ -668,6 +669,14 @@ public void SignIn(UserInfo user, bool createPersistentCookie)
                 var authCookie = HttpContext.Current.Response.Cookies[FormsAuthentication.FormsCookieName];
                 if (!string.IsNullOrEmpty(authCookie?.Value))
                 {
+                    authCookie.SameSite = SameSiteMode.Strict;
+                    if (settings.SSLEnabled)
```
Comment (mitchelsellers) [inline DNN Platform/Library/Security/PortalSecurity.cs:673]: Should we be only setting this for SSLEnabled & SSLEnforced?

Comment (Mostafa-Moafi) [reply DNN Platform/Library/Security/PortalSecurity.cs:673]: Secure cookies are sent along only with requests that travel over a secure connection; usually this means a HTTPS connection.

Comment (bdukes) [reply DNN Platform/Library/Security/PortalSecurity.cs:673]: This should use `UrlUtils.IsSecureConnectionOrSslOffload` to check if the current request is secure

**Comment by mitchelsellers** [comment]
There are situations here where existing settings are able to control this via the various cookie settings in the web.config.

Long-term, after discussion, ideally, we should be working to set these values based on the SSL Configurations to drive the setting of web.config files such as "Secure" for HttpCookies and the Auth Cookies settings via the web.config.  

A similar configuration of SameSite mode via a setting so that it is user-controllable would be preferred due to the diversity of implementations and system requirements for usage of the DNN Platform.

**Comment by mitchelsellers** [comment]
This issue was discussed at the most recent Git Approvers meeting and the end result is that we do not support this process, given the possibility of conflicting situations with other configurations based on web.config settings and how this will interact with SSL Offloading and otherwise.

We need to have a solution that will not introduce a conflict of that nature.

**Pull request closed without merging**